src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -1535,126 +1541,139 @@ public function insert_posts( $items = array() ) {
+							// Pre-set for ICS based imports
+							$organizer = false;
+							if ( ! empty( $item->organizer[ $key ]->global_id ) ) {
+								// Did we find a Post with a matching Global ID in History
+								$organizer = Tribe__Events__Aggregator__Event::get_post_by_meta(
+									'global_id_lineage',
+									$item->organizer[ $key ]->global_id
+								);
 							}
 
-							add_filter( 'tribe_tracker_enabled', '__return_false' );
-
-							// Update the Organizer
-							Tribe__Events__Organizer::instance()->update( $organizer->ID, $organizer_data );
-
-							remove_filter( 'tribe_tracker_enabled', '__return_false' );
-
-							// Tell that we updated the Organizer to the activity tracker
-							$activity->add( 'organizer', 'updated', $organizer->ID );
-						}
-					} else {
-						$organizer_id = array_search( $event['Organizer']['Organizer'], $found_organizers );
-
-						if ( ! $organizer_id ) {
-							$organizer_unique_field = $this->get_unique_field( 'organizer' );
+							if ( isset( $item->organizer[ $key ]->description ) ) {
+								$organizer_data['Description'] = $item->organizer[ $key ]->description;
+							}
 
-							if ( ! empty( $organizer_unique_field ) ) {
-								$target    = $organizer_unique_field['target'];
-								$value     = $organizer_data[ $target ];
-								$organizer = Tribe__Events__Aggregator__Event::get_post_by_meta( "_Organizer{$target}", $value );
-							} else {
-								$organizer = get_page_by_title( $event['Organizer']['Organizer'], 'OBJECT', Tribe__Events__Organizer::POSTTYPE );
+							if ( isset( $item->organizer[ $key ]->excerpt ) ) {
+								$organizer_data['Excerpt'] = $item->organizer[ $key ]->excerpt;
 							}
-						}
 
-						if ( ! $organizer_id ) {
 							if ( $organizer ) {
-								$organizer_id                      = $organizer->ID;
-								$found_organizers[ $organizer_id ] = $event['Organizer']['Organizer'];
-							}
-						}
+								$organizer_id       = $organizer_data['ID'] = $organizer->ID;
+								$event_organizers[] = $organizer_id;
 
-						// We didn't find any matching Organizer for the provided one
-						if ( ! $organizer_id ) {
-							$organizer_id = $event['EventOrganizerID'] = Tribe__Events__Organizer::instance()->create( $event['Organizer'], $this->meta['post_status'] );
-							$found_organizers[ $event['EventOrganizerID'] ] = $event['Organizer']['Organizer'];
+								$found_organizers[ $organizer->ID ] = $organizer_data['Organizer'];
 
-							// Log this Organizer was created
-							$activity->add( 'organizer', 'created', $event['EventOrganizerID'] );
+								// Here we might need to update the Organizer depending we found something based on old code
+								if ( 'retain' === $update_authority_setting ) {
+									// When we get here we say that we skipped an Organizer
+									$activity->add( 'organizer', 'skipped', $organizer->ID );
+								} else {
+									if ( 'preserve_changes' === $update_authority_setting ) {
```

Comment:
Not related to your changes in any way, but wow! Holy long-and-deeply-nested method!

Reply:
I know, it's also a really long class that will benefit from some love soon.